components/brave_wallet/browser/sns_resolver_task.cc

```diff
@@ -626,25 +650,143 @@ void SnsResolverTask::OnFetchSolRecordRegistryState(
 
   if (!account_info) {
     // No such account, use owner address.
-    task_result_.emplace(domain_name_registry_state_->owner);
+    SetAddressResult(domain_name_registry_state_->owner);
     return;
   }
 
-  sol_record_name_registry_state_ =
+  auto sol_record_name_registry_state =
       FromBase64<NameRegistryState>(account_info->data);
-  if (!sol_record_name_registry_state_) {
+  if (!sol_record_name_registry_state) {
     SetError(MakeInternalError());
     return;
   }
 
   if (auto sol_record_payload_address = ParseAndVerifySolRecordData(
-          base::make_span(sol_record_name_registry_state_->data),
+          base::make_span(sol_record_name_registry_state->data),
           sol_record_address_, domain_name_registry_state_->owner)) {
-    task_result_.emplace(*sol_record_payload_address);
+    SetAddressResult(*sol_record_payload_address);
   } else {
     // No valid address for SOL record, use owner address.
-    task_result_.emplace(domain_name_registry_state_->owner);
+    SetAddressResult(domain_name_registry_state_->owner);
+  }
+}
+
+void SnsResolverTask::FetchUrlRecordRegistryState() {
+  auto url_record_address = GetDomainKey("url." + domain_, true);
+  if (!url_record_address) {
+    SetError(MakeInternalError());
+    ScheduleWorkOnTask();
+    return;
+  }
+
+  auto internal_callback =
+      base::BindOnce(&SnsResolverTask::OnFetchUrlRecordRegistryState,
+                     weak_ptr_factory_.GetWeakPtr());
+  RequestInternal(solana::getAccountInfo(url_record_address->ToBase58()),
+                  std::move(internal_callback),
+                  solana::ConverterForGetAccountInfo());
+}
+
+void SnsResolverTask::OnFetchUrlRecordRegistryState(
+    APIRequestResult api_request_result) {
+  ScopedWorkOnSnsTask work_on_task(this);
+
+  if (!api_request_result.Is2XXResponseCode()) {
+    SetError(MakeInternalError());
+    return;
+  }
+
+  absl::optional<SolanaAccountInfo> account_info;
+  if (!solana::ParseGetAccountInfo(api_request_result.body(), &account_info)) {
+    SetError(MakeInternalError());
+    return;
+  }
+
+  if (!account_info) {
+    // No url record account, will proceed with ipfs record.
```

Comment:
why proceed with IPFS here? it seems like the user may not have opted into IPFS.

Reply:
That comment actually about next line where we don't fail with error but set a flag and proceed with ipfs record fetching in `WorkOnDomainResolveTask`.

Reply:
I think we should let user know that domain resolves to ipfs, but could not be navigated as it is disabled.
@yrliou wdyt?

Reply:
https://files.slack.com/files-pri/T04PX1BU8-F04AUF58FM3/image.png

Reply:
![image](https://user-images.githubusercontent.com/168356/201809446-dd0ab31a-5a93-44fc-8bb9-134b0a8a009e.png)

Reply:
I personally lean towards to not doing an additional check here, and just let the page fail to load if IPFS is disabled. I think it would be a more clear error as shown ^ to users.
And yes, it's correct that no requests to IPFS network will be issued until users opt in.

Reply:
sounds good to me